#!/usr/bin/env python
PACKAGE = "control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# only declaration, this is not the place to set the default values!
# look into params/longitudinal_controller instead!

# longitudinal control
gen.add("v_min_cc",                        double_t, 0, "minimum velocity",      0,   0, 5)
gen.add("v_max_cc",                        double_t, 0, "maximum velocity",      0,   0, 5)

gen.add("a_krit",                       double_t, 0, "maximum side acceleration", 0,    0, 20)
gen.add("lookahead_cc_t",                  double_t, 0, "lookahead for speedcontrol",  0,  0, 1)
gen.add("v_extrapolation",                  double_t, 0, "assumed maximal drivable speed at end of path",  0,  0, 4)

gen.add("max_acc",            			double_t, 0, "max acceleration",   0,  0, 10)
gen.add("max_dec",            			double_t, 0, "max deceleration",   0,  0, 100)


exit(gen.generate(PACKAGE, "longitudinal_controller", "LongitudinalController"))
 
